packages/documentation/src/pages/architecture/unit-testing.mdx

```diff
@@ -0,0 +1,108 @@
+## How do unit tests work?
+
+|Test Type|Description|File Format|Where|When|Libraries|
+|---|---|---|---|---|---|
+| Unit | Test for individual functions or components | *.unit.spec.js | Any .unit.spec.js file located in the `/src` folder. Typically located in a directory close to the code being tested | Locally through npm script commands<br/><br/>Jenkins build (Unit)<br/><br/>After merging to master | [mocha.js](https://mochajs.org/)<br/><br/>[chai.js](https://www.chaijs.com/)<br/><br/>[chai-as-promised](https://github.com/domenic/chai-as-promised)<br/><br/>[sinon.js](https://sinonjs.org/)<br/><br/>[enzyme](https://airbnb.io/enzyme/)<br/><br/>[enzyme-adapter-react-16](https://www.npmjs.com/package/enzyme-adapter-react-16)<br/><br/>[react-dom](https://www.npmjs.com/package/react-dom)<br/><br/>[react-dom/test-utils](https://reactjs.org/docs/test-utils.html)<br/><br/>[react-test-renderer](https://www.npmjs.com/package/react-test-renderer)<br/><br/>[jsdom](https://airbnb.io/enzyme/docs/guides/jsdom.html) |
```

Comment:
I like having a table for the different types of tests, but this one is pretty big. Is there a way to make it more compact, by cutting down on descriptive text (or moving that text to a different part of the doc)? I think it's most useful if this table is a summary that gets you information quickly.

Reply:
@jbalboni I removed the descriptions. If a developer doesn't know what the tests are, they can just google it. I think most developers should know what they are.

Reply:
I still feel like it's hard to scan. Maybe @rianfowler feels differently, but I'm reconsidering my liking of a table here. We already have a table that gives some basics: https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/common-tasks/test

Maybe we should go back to just a narrative instead of a table. And focus on explaining what's happening in each type of test.

Sorry for the back and forth on this, I realized I was pushing towards a table that we already have in the common tasks section, which made me rethink the idea of a table here entirely.

Reply:
@jbalboni No problem, I will change it up so it is all sectioned out but without the tables.

Reply:
I think the goal of this document is to dig into what our tests are doing to give someone a solid mental model for how each of the suites executes. I believe this is mostly for people who are troubleshooting issues with the tests but we could break troubleshooting off into another section or ticket. 

I think we link to the getting started document with something like `[common test commands](https://department-of-veterans-affairs.github.io/veteran-facing-services-tools/getting-started/common-tasks/test/#test-commands)`. We could follow the same table formatting to document the remaining commands in our package.json (maybe omitting the test all/ test one columns to simplify it). Maybe a separate table or a separate document for each test type.

I think for each type of test (unit, lint, end to end, accessibility) it would be good to bullet list the process these tests go through. e.g. 

At a high level our unit test follows this process: 

- Mocha builds target unit test files in the same environment 
  - Babel is used to transpile the unit tests 
- CHOMA (_not sure if this is a Mocha extension or something else_) randomizes the test execution 
- Unit tests are executed serially 
- Mocha uses the _?_ reporterer to output the test results 
- _example of a success report_
- _exmaple of a failure report_

Reply:
Just to emphasize: I think defining the goal of this document will make it easier to edit.